src/common/mempool.cc

```diff
@@ -42,13 +42,16 @@ const char *mempool::get_pool_name(mempool::pool_index_t ix) {
 void mempool::dump(ceph::Formatter *f)
 {
   stats_t total;
+  f->open_object_section("mempool"); // we need (dummy?) topmost section for 
+				     // JSON Formatter to print pool names. It omits them otherwise.
```

Comment:
@ifed01 i think we could add a dummy section here, like
```c++
f->open_object_section("mempool");
f->open_array_section("pools");
for (...) {
  ...
}
f->close_section();
f->dump_object("total", total);
f->close_section();
```

as the pools are of the same type (`pool_t`), while `total` is a `stats_t`. what do you think?

Reply:
makes sense. will do

Reply:
@tchaikov - unfortunately it looks like top level blocks inside an array can't have names as well - hence we either need yet another dummy nested level or should print pool names as a field within a block not as a block name. Consider:
{
"pools" : [
   {
     "pool1" : {
        "a":"1"
      }
   },
   {
     "pool2" : {
       "a":"1"
     }
   }
]
}
vs.
{
"pools" : [
 {
   "name" : "pool2"
   "a":"1"
 },
 {
   "name" : "pool2"
   "a":"1"
 }
]
}
Not sure adding array section worths such changes... Please note that these 'dummy' block names will be visible in other formats hence wasting the output...

Reply:
no, elements in an array does not have their names. so, right, the output would look like the latter. 

IMHO, the problem of `mempool::dump()` are two folded:

1. no top level data structure. that's what your change is addressing
2. "total" is not a pool, but it is printed in line with other pools.